### PR TITLE
Navmap UI enhancements

### LINKED ISSUE #23584: Station Map Text Low Contrast
## Description
image

some of the dark red text have poor visibility

### PR DESCRIPTION
**Pull request by Errant-4**
## About the PR
tldr:
Map label (beacon) presentation has been improved and should remain consistent across all configurations and scenarios. If the labels didn't overlap when they were mapped, they won't overlap under any circumstances. Wysiwyg
Using Examine on the map screen now toggles the labels (the checkbox also still works), since they don't get "out of the way" anymore when zoomed in

- [X]  Beacon labels on the navmap now scale with the zoom level. They remain consistent to how they looked in relation to the map and each other when zooming the map. Gone is the overlap of doom.
- ~~Interface hooks created for the player to adjust Font size (before scaling) and toggle the map labels between normal and bold~~ Removed
- ~~Maybe add the same stuff for label background opacity?~~ Not needed, the adjusted one should be good enough plus you can zoom in on labels now

- [X]  Beacon text is now bold and the background is less transparent for improved contrast
- [X] Crew Monitor and Power Monitor interfaces now use their custom tile colors as background for ui elements, instead of the default tile color from the Station Map

- [X] ~~In addition to it's current function of Unselecting, Right clicking~~ on the map screen now ~~also~~ toggles station beacons ~~(implemented but for some reason it only works on the Crew Monitor and Power Monitor, not the Station map. Someone, anyone, please tell me why)~~
-- [X]  Fix ~~right click~~ toggle not working on Station Map UI
-- [X] Changed Right Click to Examine

- [X] The zoom percentage counter, which was not grounded in anything the player could comprehend and showed the opposite of what numbers mean, now shows the actual zoom level compared to the default view
- ~~Something something too many Warp Points~~ Out of scope

Fixes #23584

## Media
map_pr

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Errant
- fix: Map labels now adjust to zoom level and UI scale, retaining a consistent look.
- tweak: Map labels now have better contrast over walls.
- tweak: Examining (shift-click) the map screen now toggles map labels on/off.

### CODE REVIEW

File: Content.Client/Pinpointer/UI/NavMapControl.cs [hunk 1]
```diff
@@ -367,15 +374,21 @@ protected override void Draw(DrawingHandleScreen handle)
         if (_beacons.Pressed)
         {
             var rectBuffer = new Vector2(5f, 3f);
+            var cache = IoCManager.Resolve<IResourceCache>();
+            Font font;
```
Comment (metalgearsloth) [inline Content.Client/Pinpointer/UI/NavMapControl.cs:378]: Store the dependency rather than resolving the threadlocal every frame please.

File: Content.Client/Pinpointer/UI/NavMapControl.cs [hunk 2]
```diff
@@ -236,6 +240,9 @@ protected override void KeyBindUp(GUIBoundKeyEventArgs args)
             // Clear current selection with right click
             if (TrackedEntitySelectedAction != null)
                 TrackedEntitySelectedAction.Invoke(null);
+
+            // Toggle beacon labels
+            _beacons.Pressed = !_beacons.Pressed;
         }
```
Comment (chromiumboy) [inline Content.Client/Pinpointer/UI/NavMapControl.cs:246]: It's not a good idea to have two different functions tied to the same player input. I'd remove this

Comment (Errant-4) [reply Content.Client/Pinpointer/UI/NavMapControl.cs:246]: I'll see if I can find a better keybind for the toggle but if I dont, then I would rather remove the unselect function. When does anyone ever need/try to untarget something on the map view? Does anyone have any recollection of ever using that function?

Comment (Errant-4) [reply Content.Client/Pinpointer/UI/NavMapControl.cs:247]: I moved it to Examine

File: Content.Client/Pinpointer/UI/NavMapControl.cs (hunk 1 above)
Comment (Errant-4) [reply Content.Client/Pinpointer/UI/NavMapControl.cs]: Ok I think I got it

**Comment by Errant-4** [comment]
All done
